renderer_test.go

```diff
@@ -41,6 +41,27 @@ func TestWriteColorOverride(t *testing.T) {
 	assert.NotContains(t, renderer.string(), "<#ff5733>")
 }
 
+func TestWriteColorOverrideBackground(t *testing.T) {
+	renderer := &Renderer{
+		Buffer: new(bytes.Buffer),
+	}
+	text := "This is white, <,#000000>this is black</>, white again"
+	renderer.init("pwsh")
+	renderer.write("#193549", "#ff5733", text)
+	assert.NotContains(t, renderer.string(), "000000")
+}
+
+func TestWriteColorOverrideBoth(t *testing.T) {
```

Comment:
can you add a test with 16 colors eg: `"This is white, <black,white>this is black</>, white again"`

Reply:
I added some additional unit tests to ensure the 16 color format is also properly handled, let me know if this is what you're looking for!

Reply:
I'm going to do a quick checkout of your branch as I feel the regex could be updated. But I'm not sure so I'll do a quick debug ;-)

Reply:
I'll refactor the regex later. It's out of scope for this one. Well done!